### PR TITLE
Default impl `BlockBody::recover_signers`

### PR DESCRIPTION
**Pull request by emhane**
Ref https://github.com/paradigmxyz/reth/issues/12575

### CODE REVIEW

File: crates/primitives-traits/src/block/body.rs [hunk 1]
```diff
@@ -29,8 +45,17 @@ pub trait BlockBody:
     + MaybeArbitrary
 {
     /// Ordered list of signed transactions as committed in block.
-    type Transaction: Transaction;
+    type Transaction: SignedTransaction;
```
Comment (mattsse) [inline crates/primitives-traits/src/block/body.rs:48]: can we do this separately?

**Review by mattsse — COMMENTED** [review]
hmm, don't think we should make this the default impl, because this embeds rayon at the root of the abstraction

File: crates/primitives-traits/src/block/body.rs (hunk 1 above)
Comment (emhane) [reply crates/primitives-traits/src/block/body.rs:48]: https://github.com/paradigmxyz/reth/pull/12903

**Comment by emhane** [comment]
closing as this was included in https://github.com/paradigmxyz/reth/pull/12794

**Pull request closed without merging**